synapse/storage/databases/main/roommember.py

```diff
@@ -1129,12 +1136,14 @@ def f(txn):
 
 @attr.s(slots=True)
 class _JoinedHostsCache:
-    """Cache for joined hosts in a room that is optimised to handle updates
-    via state deltas.
-    """
+    """The cached data used by the `_get_joined_hosts_cache`."""
 
     # Dict of host to the set of their users in the room at the state group.
     hosts_to_joined_users = attr.ib(type=Dict[str, Set[str]], factory=dict)
+
+    # The state group `hosts_to_joined_users` is derived from. Will be an object
+    # if the class is newly created or if the state is not based on a state
+    # group.
```

Comment:
does the object mean anything, or is it just a sentinel to ensure it is never equal to anything else? (in which case, say so?)